src/diffusers/pipeline_utils.py

```diff
@@ -175,9 +175,9 @@ def save_pretrained(self, save_directory: Union[str, os.PathLike]):
                         break
                 if save_method_name is not None:
                     break
-
-            save_method = getattr(sub_model, save_method_name)
-            save_method(os.path.join(save_directory, pipeline_component_name))
+            if save_method_name:
```

Comment:
From the type hint it seems like `save_method_name` has to be either a tsring or a path so this statement will always be true and thus I don't think it's necessary.  

Why should we maybe this change? Could you give more context on the PR :-) ?